basilisk/singleton_instance/posix_singleton_instance.py

```
@@ -0,0 +1,201 @@
+		Returns:
+			The PID of the existing instance, or None if no instance is running.
+		"""
+		# POSIX file-based approach with proper lock checking
+		if not os.path.exists(self.lock_file_path):
+			return None
+
+		pid = self._get_lock_holder_pid()
+		if pid is None:
+			return None
+
+		# Check if the process is still alive
```

Flake8 (7.2.0)

[warning] indentation contains tabs

(W191)